### PR TITLE
Second update

### PR DESCRIPTION
**Pull request by petersen-f**
Implemented midi input and psychopy to display notes and get feedback

### CODE REVIEW

**Comment by vandenman** [comment]
@AlexisDaniels & @JuliusMaerz a lot of files changed here! You can use the file filter to deselect file types that you don't want to look at (I'd only look at python files). You can find that here: 
image

image

File: piano_package/midi_piano.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+from psychopy.core import wait, Clock, quit
+from psychopy.hardware.keyboard import Keyboard
+
+# midi functions
+from pygame import midi
+
+
+class midi_piano:   
+    '''
+    This class defines the properties of the midi piano and inicates the location of the note pictures (and maybe sound files).
```
Comment (AlexisDaniels) [inline piano_package/midi_piano.py:13]: small typo in line 13 "indicates"

File: piano_package/midi_piano.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    This is important as different midi pianos have different key_ids and inidcations of whether a button is pressed.
```
Comment (AlexisDaniels) [inline piano_package/midi_piano.py:14]: small typo in line 14, again with "indications"

**Review by AlexisDaniels — COMMENTED** [review]
Great stuff! I was not able to go through it with much ease because there are lots of dependencies between lots of files. I left you a couple of comment where I spotted typos (they were comments, so no repercussions on the code)

Looking forward for your final app

**Review by JuliusMaerz — COMMENTED** [review]
In general this all looks very nice. I had to read it a few times to figure out how the different documents were connected, but now it´s pretty clear to me.

File: Incremental Scripts/5_midi_and_notes_to_psychopy.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+key_bindings = read_csv("./key_bindings.csv",index_col=0)
+
+
+#condition_df = key_bindings.sample(frac=1)
+condition_df = key_bindings
+midi.init()
+
+input = midi.Input(1)
+
```
Comment (JuliusMaerz) [inline Incremental Scripts/5_midi_and_notes_to_psychopy.py:53]: Maybe some comments/documentation about the code from line 53 to 90 would be helpful to understand what is going on. But maybe I´m just too slow to fully understand it :D

**Pull request merged**